packages/x-date-pickers/src/YearCalendar/PickersYear.tsx

```diff
@@ -119,44 +124,52 @@ export const PickersYear = React.memo(function PickersYear(inProps: PickersYearP
     'aria-current': ariaCurrent,
     // We don't want to forward this prop to the root element
     yearsPerRow,
+    slots,
+    slotProps,
     ...other
   } = props;
 
   const ref = React.useRef<HTMLButtonElement>(null);
   const classes = useUtilityClasses(props);
 
   // We can't forward the `autoFocus` to the button because it is a native button, not a MUI Button
-  React.useEffect(() => {
+  useEnhancedEffect(() => {
```

Comment:
The month component was using one effect, the year component was using the other effect, I tried to reconcile the two.

For me, the goal of `useEnhancedEffect` is that, on re-renders it runs before the visual changes are applied, so it can avoid some flickering.
Not sure it has any impact here though

Reply:
Ok, thanks for clarifying. That's a fair point, if it's not a problem on `PickersMonth`, let's unify the approach. 👍